### PR TITLE
[Reopen to 'current' branch] Make location parameter inline to fix issue 1508

### PR DESCRIPTION
**Pull request by felixwa**
Make location parameter inline, so that AutoRest will generate it as a
parameter of the method instead of a property of Client class.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by felixwa** [comment]
Previous PR was opened to 'master' now open it to 'current' as @dsgouda suggested.

**Comment by felixwa** [comment]
To fix Issue #1508

**Review by dsgouda — APPROVED** [review]
Thanks, LGTM

**Comment by olydis** [comment]
This fix could have been way simpler! Just add `"x-ms-parameter-location": "method"` (see https://github.com/Azure/azure-rest-api-specs/blob/current/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/2016-06-01/registeredIdentities.json#L73) to the parameter declaration - it's no big deal here since you only used the parameter once, but before unrolling *multiple* referenced to the same parameter, definitely use `x-ms-parameter-location`.

**Comment by dsgouda** [comment]
On second thoughts I agree with @olydis , if you do ever need to use `resourceGroupNameParamater` in some other operation, you can simply reference this global definition (something we'd recommend anyway)

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/2017-04-18/cognitiveservices.json [hunk 1]
```diff
@@ -1028,13 +1032,6 @@
       "required": true,
       "type": "string",
       "description": "Version of the API to be used with the client request. Current version is 2017-04-18"
-    },
-    "locationParameter": {
-      "name": "location",
-      "in": "path",
-      "description": "Resource location.",
-      "required": true,
-      "type": "string"
```
Comment (dsgouda) [inline specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/2017-04-18/cognitiveservices.json]: set `x-ms-parameter-location` here instead, should be a one line change

**Review by dsgouda — CHANGES_REQUESTED** [review]
Requesting a minor change

**Comment by felixwa** [comment]
Okay, now use "x-ms-parameter-location": "method" to decorate the parameter.

**Review by olydis — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]

**Pull request merged**